### PR TITLE
EOBS-1889_partial_reason_popup

### PR DESCRIPTION
**Pull request by Nayira**
EOBS-1889_partial_reason_popup

# Features Added/Changed:
- Refactored step confirm_calculated_value to include regex match for… partial observations submission
- Edited gherkin on 3in1_risk_score_calculation and submit_news_obs_risk_score_calculation to match refactored step
- Added submit_partial_news_obs feature
- Added string.py file to contain regex list - this contains a 'disable' line for 'anomalous-backslash-in-string' due to regex syntax

# Feature file checks:
- [x] JIRA ticket reference number
- [x] Scenarios covered listed
Notes: At the time of PR, this 'Story' does not exist in JIRA - need to edit in the future

# Possible issues:
Adding regex checks for Score and Clinical Risk on the popup. Might need to refactor to include other obs types and/or include regex for full obs for better coverage.

# General Checks:
- [x] Linting passing with score 10/10 (Happy Travis!)
- [x] All Tests passing on local run
- [x] All commented out code has a reason for it documented
- [x] All methods have a detailed docstring / changed methods have updated docstring
- [x] Documentation up to date
- [x] Timings on expected conditions reviewed
- [x] Deprecated selectors reviewed
- [x] Code re-usability reviewed

### CODE REVIEW

File: steps/mobile_common_actions.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 from liveobs_ui.selectors.mobile.get_selector_by_lookup import \
     get_element_selector
+from .strings import get_string_regex
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Would it be better to move the regex matching functionality into the selenium library?

File: steps/mobile_common_actions.py [hunk 2]
```diff
@@ -215,21 +216,36 @@ def submit_the_form(context):
...
 
 
-@then('the {value_to_check} submitted is {expected_value}')
-def confirm_calculated_value(context, value_to_check, expected_value):
+@then('the {obs_type} {value_to_check} submitted is {expected_value}')
+def confirm_calculated_value(
+        context, obs_type, value_to_check, expected_value):
     """
-    Gets and verifies a value (clinical risk or score) displayed in the
+    Gets and verifies a value (clinical risk or score) displayed in the partial
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Added partial to docstring but if I remember correctly it covers both full and partial don't it?

File: steps/mobile_common_actions.py [hunk 3]
```diff
@@ -215,21 +216,36 @@ def submit_the_form(context):
...
     submission confirmation popup
     :param context: behave driver
-    :param value_to_check: The value 'name' to look for. Can be Clinical Risk,
-     NEWS score or GSC score. Refers to a specific locator in the page, by text
-    :param expected_value: the value expected
+    :param obs_type: specifies using regex or locators to match against the
+    value_to_check parameter
+    :param value_to_check: The value 'name' to look for. Refers to a specific
+    locator in the page, by text
+    :param expected_value: the value expected. Example: 'No' Clinical Risk or
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: I wouldn't use the `param` for the detailed examples of input. I'd use the main docstring body as when it's turned into docs it'll read better http://liveobs-selenium.readthedocs.io/en/latest/_modules/liveobs_ui/page_object_models/common/base_liveobs_page.html#BaseLiveObsPage.fill_out_form

File: steps/mobile_common_actions.py [hunk 4]
```diff
@@ -247,4 +263,38 @@ def confirm_observation_submission(context, obs_name):
     modals = modal_page.get_open_modals()
     submit_modal = modals[0]
     assert modal_page.get_modal_title(submit_modal) == \
-        'Successfully Submitted {} Observation'.format(obs_name)
+        'Successfully Submitted {} Observation'.format(obs_name), \
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Could use regex for this so can match on both full and partial obs?

File: steps/strings.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+REGEXES = {
+    'Clinical Risk': PARTIAL_NEWS_CLINICAL_RISK,
+    'Score': PARTIAL_NEWS_SCORE,
+    'OBS': OBS_TITLE
+}
+
+
+def get_string_regex(nice_human_name):
+    """
+    Get a string regex by supplying a nicer name or something, I dunno
```
Comment (Gimpneek) [inline steps/strings.py]: What the hell is this docstring? What kind of person would write such a thing 😹

File: steps/mobile_common_actions.py (hunk 1 above)
Comment (Nayira) [reply steps/mobile_common_actions.py]: Discussed it to move later - as atm it doens't seem we can use it anywhere else aside from automation

File: steps/mobile_common_actions.py (hunk 4 above)
Comment (Nayira) [reply steps/mobile_common_actions.py]: Will work on this on a separate branch :)

File: steps/strings.py (hunk 5 above)
Comment (Nayira) [reply steps/strings.py]: I know right? Not sure who wrote it :suspect:

**Pull request merged**